lib/google/gax/configuration.rb

```diff
@@ -231,16 +235,16 @@ def reset! key = nil
         if key.nil?
           @values.each_key { |k| reset! k }
         else
-          key = key.to_sym
-          if @defaults.key? key
-            @values[key] = @defaults[key]
-            @values[key].reset! if @validators[key] == SUBCONFIG
+          key = ::Kernel.String(key).to_sym
```

Comment:
All keys are converted to symbols today. The current code uses `key.to_s.to_sym`, but I replaced it with `::Kernel.String(key).to_sym` so that an error would be raised if a key object was ever passed in that wasn't able to be converted to a string.